src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -881,12 +884,51 @@ private OpenApiSchema GenerateSchemaFromFormParameters(
                 Required = new SortedSet<string>(requiredPropertyNames)
             };
         }
+        private static IList<ApiResponseType> GetResponseTypes(ApiDescription apiDescription)
+        {
+#if NET7_0_OR_GREATER
+            var supportedResponseTypes = new List<ApiResponseType>(apiDescription.SupportedResponseTypes);
+            if (apiDescription.ActionDescriptor is ControllerActionDescriptor controllerActionDescriptor)
+            {
+                var returnType = UnwrapTask(controllerActionDescriptor.MethodInfo.ReturnType);
+                if (typeof(IEndpointMetadataProvider).IsAssignableFrom(returnType))
+                {
+                    var populateMetadataMethod = returnType.GetMethod("Microsoft.AspNetCore.Http.Metadata.IEndpointMetadataProvider.PopulateMetadata", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
I wonder if it's best to make this specific so we couldn't be broken in the future by any other overloads being added.

Also, why does it need `NonPublic` - the interface method should be public.

Reply:
Just with static is enough, because it's marked as static